### DISCUSSION TITLE
Data privacy and sharing

### DISCUSSION THREAD

**Discussion by McFly0101** [seleniumbase/SeleniumBase#2227] [Q&A] [answered]
Hi, I am learning to use this, but I am concern on data privacy and sharing issues. May I know does the application or plug in make any network requests or send data externally when/after processing?

**Comment by mdmintz** [comment, marked as answer]
Hi, If you already have a driver that matches your browser (eg. `chromedriver` 118 for Chrome 118), then no calls are made.

So assuming you already have your matching driver and you've already cloned/installed SeleniumBase, you can disconnect your internet connection and run the offline tests from this folder: [SeleniumBase/examples/offline_examples](https://github.com/seleniumbase/SeleniumBase/tree/master/examples/offline_examples).

Additionally, there's a pytest command-line option: `--driver-version="keep"`, that you can use to keep using your existing chromedriver, even if it doesn't match your Chrome version. (Eg. If you have chromedriver 115 and Chrome 118, then using `--driver-version="keep"` means no driver will be downloaded.)

When drivers are downloaded automatically, they come from the source:
* `chromedriver`:
-- version 114 and older: https://chromedriver.chromium.org/downloads
-- version 115 and newer: https://googlechromelabs.github.io/chrome-for-testing/
* `msedgedriver`: https://developer.microsoft.com/en-us/microsoft-edge/tools/webdriver/
* `geckodriver`: https://github.com/mozilla/geckodriver/releases

**Reply by McFly0101** [reply]
Appreciate your quick reply☺️
